recipes/C-iTRACE/recipe.py

```diff
@@ -0,0 +1,24 @@
+from pangeo_forge_recipes.patterns import ConcatDim, FilePattern, MergeDim
+from pangeo_forge_recipes.recipes.xarray_zarr import XarrayZarrRecipe
+
+# C-iTRACE covers a number of proxies
+variables = ['d18O']
+
+def make_url(time, variable):
+    url = 'https://gdex.ucar.edu/dataset/204_ajahn/file/ctrace.decadal.{variable}.nc'.format(variable=variable)
+    return url
+
+# there is only one, but notably ~6 GB file covering the full timeseries
+time_concat_dim = ConcatDim("time", [0], nitems_per_file=2200)
+pattern = FilePattern(make_url,
+                      time_concat_dim,
+                      MergeDim(name="variable", keys=variables))
```

Comment:
```suggestion
                      MergeDim(name="variable", keys=variables),
                      file_type="netcdf3")
```
@jordanplanders, I'm guessing these input files may be in NetCDF3 (i.e. classic NetCDF) format?

If so, could you accept this suggestion? If not, we'll need to do a bit more digging on this.

Reply:
@jbusecke and I previously worked out how to check if a file served over ESGF is in NetCDF3 format. That function is here, and could probably be extracted/adapted to check this:

https://github.com/jbusecke/pangeo-forge-esgf/blob/dc6ca0a72486a70813042caeba1ce13742606745/pangeo_forge_esgf/dynamic_kwargs.py#L171-L183

Reply:
@cisaacstern The metadata indicates NCO version 4.9.5, so I think we'll have to look elsewhere. I've encountered another case which falsely triggered an engine error, but the details of what fixes it are a bit hazy. I'll dig around and see what I can learn.

Reply:
@jordanplanders, sounds good. To phrase the current blocking question perhaps more clearly, if you download this (6 gb) input file, what value do you have to pass for `engine`, to get this to open? 

```python
import xarray as xr

local_fpath = "place/where/this/file/was-saved.nc"
ds = xr.open_dataset(local_fpath="?")
```

Reply:
@jordanplanders, just wanted to mention that the error _does strongly suggest_ that this file is in NetCDF3 format. So much so that I would be somewhat inclined to guess that the metadata might be wrong? The code from @jbusecke that I linked above checks for NetCDF format version based on reading bytes out of the file (rather than via plain text metadata, which could conceivably be incorrect), so would be a more conclusive test for format version. It might be worth running this check just to make sure the metadata is not incorrect.

Reply:
@cisaacstern Well, I'll be!  The following loads the data as expected:
```
file_name = "ctrace.decadal.d18O.nc"
ds = xr.open_dataset(Path('data') /file_name, engine='netcdf4', decode_coords=True)
in_memory = ds.isel(time=slice(1000, 1009)).load()
```
But!  I added the line about `file_type="netcdf3"` and it worked!  I also tidied up a few other things, and adjusted a few parameters to make it *much* faster (at least locally) so I'll do a few more checks and push changes.

There are a number of other proxies in the same suite that I'll eventually want to add.  Is it best to have one c-itrace recipe with a set of variables which handles the merge, or to have one recipe per proxy and leave it to the intrepid user to pull them together or not as they wish?

Reply:
Do they all have identical dimensions? Then I would always argue for merging them.